### ISSUE TITLE
crc32_iscsi_00 deal with null buffer

### ISSUE THREAD

**Issue by changchengx** [intel/isa-l#159]
In the distributed storage system Ceph(github.com/ceph/ceph), it needs to deal with the case when the input buffer is NULL while the length isn't zero: https://github.com/ceph/ceph/blob/master/src/test/common/test_crc32c.cc#L358
On IA platform, the crc32 is implemented by:
1. https://github.com/ceph/ceph/blob/master/src/common/crc32c_intel_fast_asm.s
    This file is copied from : https://github.com/intel/isa-l/blob/master/crc/crc32_iscsi_00.asm
2. https://github.com/ceph/ceph/blob/master/src/common/crc32c_intel_fast_zero_asm.s
    This file changes something from crc32c_intel_fast_asm.s to be renamed as crc32c_intel_fast_zero_asm.s to deal with the case when the input buffer is NULL.

Could isa-l adds support to deal with NULL buffer input crc32c case?

**Comment by changchengx** [comment]
@gbtucker ping

**Comment by gbtucker** [comment]
I think the null buffer crc case is fairly particular to Ceph so I don't see adding it to all our CRCs.  I'd suggest that Ceph check for NULL and call the zerro buffer routine in Ceph codebase and when non-null, call our multibinary function crc32_iscsi().  This way you can also get the new vclmul version.

**Comment by changchengx** [comment]
@gbtucker 
In Ceph, there's a particular version to calculate null buffer(should be zero) crc: https://github.com/ceph/ceph/blob/master/src/common/crc32c_intel_fast_zero_asm.s
For the normal non-null case, Ceph use "crc32_iscsi_00" function. Do you mean that Ceph should call "crc32_iscsi" function for better performance?

**Comment by gbtucker** [comment]
@changchengx, yes the crc32_iscsi() will dispatch to one of 4 different versions based on instruction availability.  It might be easier to just call this than to pull multiple versions in and dispatch yourself.
